iDiff/utils/tar_utils.go

```diff
@@ -15,6 +15,32 @@ import (
 	"github.com/docker/docker/pkg/system"
 )
 
+// ImageToDir converts an image to an unpacked tar and creates a representation of that directory.
+func ImageToDir(img string) (string, string, error) {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		return "", "", err
+	}
+	tarPath, err := ImageToTar(cli, img)
+	if err != nil {
+		return "", "", err
+	}
+	//tarPath := filepath.Join(img, ".tar")
+	err = ExtractTar(tarPath)
+	if err != nil {
+		return "", "", err
+	}
+	os.Remove(tarPath)
+	path := strings.TrimSuffix(tarPath, filepath.Ext(tarPath))
+	jsonPath := path + ".json"
```

Comment:
ahh right, sorry this is my bad. didn't register that you're just adding the extension to the end of the file, not the whole filename. this is fine :+1: